src/Functions/formatReadableTimeDelta.cpp

```diff
@@ -116,12 +134,53 @@ class FunctionFormatReadableTimeDelta : public IFunction
             max_unit = Minutes;
         else if (maximum_unit_str == "seconds")
             max_unit = Seconds;
+        else if (maximum_unit_str == "milliseconds")
+            max_unit = Milliseconds;
+        else if (maximum_unit_str == "microseconds")
+            max_unit = Microseconds;
+        else if (maximum_unit_str == "nanoseconds")
+            max_unit = Nanoseconds;
         else
             throw Exception(ErrorCodes::BAD_ARGUMENTS,
                 "Unexpected value of maximum unit argument ({}) for function {}, the only allowed values are:"
-                " 'seconds', 'minutes', 'hours', 'days', 'months', 'years'.",
+                " 'nanoseconds', 'microseconds', 'nanoseconds', 'seconds', 'minutes', 'hours', 'days', 'months', 'years'.",
                 maximum_unit_str, getName());
 
+        if (minimum_unit_str.empty() || minimum_unit_str == "seconds") // Set seconds as min_unit by default not to ruin old use cases
+            min_unit = Seconds;
+        else if (minimum_unit_str == "years")
+            min_unit = Years;
+        else if (minimum_unit_str == "months")
+            min_unit = Months;
+        else if (minimum_unit_str == "days")
+            min_unit = Days;
+        else if (minimum_unit_str == "hours")
+            min_unit = Hours;
+        else if (minimum_unit_str == "minutes")
+            min_unit = Minutes;
+        else if (minimum_unit_str == "milliseconds")
+            min_unit = Milliseconds;
+        else if (minimum_unit_str == "microseconds")
+            min_unit = Microseconds;
+        else if (minimum_unit_str == "nanoseconds")
+            min_unit = Nanoseconds;
```

Comment:
Maybe factor it out to a function? RN you have two identical pieces, one for max and one for min units.